tls.go

```diff
@@ -0,0 +1,268 @@
+	CertificateMap string `long:"certificate-map" description:"A file mapping server names to certificates"`
+	// TODO: directory? glob?
+	RootCAs string `long:"root-cas" description:"Set of certificates to use when verifying server certificates"`
+	// TODO: format?
+	NextProtos              string `long:"next-protos" description:"A list of supported application-level protocols"`
+	ServerName              string `long:"server-name" description:"Server name used for certificate verification and (optionally) SNI"`
+	VerifyServerCertificate bool   `long:"verify-server-certificate" description:"If set, the scan will fail if the server certificate does not match the server-name, or does not chain to a trusted root."`
+	// TODO: format? mapping? zgrab1 had flags like ChromeOnly, FirefoxOnly, etc...
+	CipherSuite      string `long:"cipher-suite" description:"A list of cipher suites to use."`
+	MinVersion       int    `long:"min-version" description:"The minimum SSL/TLS version that is acceptable. 0 means that SSLv3 is the minimum."`
+	MaxVersion       int    `long:"max-version" description:"The maximum SSL/TLS version that is acceptable. 0 means use the highest supported value."`
+	CurvePreferences string `long:"curve-preferences" description:"A list of elliptic curves used in an ECDHE handshake, in order of preference."`
+	NoECDHE          bool   `long:"no-ecdhe" description:"Do not allow ECDHE handshakes"`
+	// TODO: format?
+	SignatureAlgorithms string `long:"signature-algorithms" description:"Signature and hash algorithms that are acceptable"`
+	HeartbeatEnabled    bool   `long:"heartbeat-enabled" description:"If set, include the heartbeat extension"`
+	DSAEnabled          bool   `long:"dsa-enabled" description:"Accept server DSA keys"`
+	// TODO: format?
+	ClientRandom string `long:"client-random" description:"Set an explicit Client Random (base64 encoded)"`
+	// TODO: format?
+	ClientHello string `long:"client-hello" description:"Set an explicit ClientHello (base64 encoded)"`
+}
+
+func getCSV(arg string) []string {
+	// TODO: Find standard way to pass array-valued options
+	split := strings.Split(arg, ",")
+	ret := make([]string, len(split))
+	for i, v := range split {
+		ret[i] = strings.Trim(v, " \t")
+	}
+	return ret
+}
+
+func (t *TLSFlags) GetTLSConfig() (*tls.Config, error) {
+	var err error
+
+	// TODO: Find standard names
+	cipherMap := map[string][]uint16{
+		"dhe-only":        tls.DHECiphers,
+		"ecdhe-only":      tls.ECDHECiphers,
+		"exports-dh-only": tls.DHEExportCiphers,
+		"chrome-only":     tls.ChromeCiphers,
+		"chrome-no-dhe":   tls.ChromeNoDHECiphers,
+		"firefox-only":    tls.FirefoxCiphers,
+		"firefox-no-dhe":  tls.FirefoxNoDHECiphers,
+		"safari-only":     tls.SafariCiphers,
+		"safari-no-dhe":   tls.SafariNoDHECiphers,
+	}
+
+	ret := tls.Config{}
+
+	if t.Time != "" {
+		// TODO: Find standard time format
+		var baseTime time.Time
+		baseTime, err = time.Parse("20060102150405Z", t.Time)
```

Comment:
Why do we need `baseTime`?

Reply:
I think that was mostly for readability, but it would probably help if there weren't 7 unrelated lines between where it's defined and where it's used

 - [ ] Move down